### DISCUSSION TITLE
Introduce special mark for cross config references

### DISCUSSION THREAD

**Discussion by patrickpichler** [Dynatrace/dynatrace-configuration-as-code#208] [Archive] [answered]
As of right now we cannot really distinguish between simple values and cross config 
references. The only way to see if a value is a cross reference is by checking them
to end with either `.id` or `.name`. With this approach every value ending with either
of those will get interpreted as a reference. 

This approach right now does not scale. 

Once we decide to add more and more cross references options, this will get more 
and more complicated.

I propose to use some special format to mark cross config references. 
Something like `$`.

So if we now want to cross reference the id of another config we simply write
`$/other/dashboard.id` or `$other/dashboard.id`. 

That way we make cross references explicit. 

As this would be a breaking change, we should think about introducing a helper
command for migrating to this new convention. 

It would also make sense to switch all paths to an os interdependent format (#21)
while doing this, as this would also break compatibility.

**Comment by tatomir146** [comment, marked as answer]
Just a though: would it be possible to use go templates for this? We already provide environment variables `{{ .Env.MyVar}}`. Wouldn't it be nice to have something like this `{{ .Var.project.dashboard.whatever}}`?
